### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by jason-fox**
## Description

- Add `SHELL` command
- Add Metadata `LABELS` following opencontainers.org
- Add maintainer `LABEL`
- Install locales and tzdata as part of the base image.

Related #3664

## Motivation and Context

I want to reduce the GitHub Docker container action to just some scripting hooks. At the moment I need to do the following installs (based on my unit-test GitHub Action experiences):

```bash 
apt-get install -y --no-install-recommends openjdk-11-jre ant curl unzip git locales tzdata
``` 

Most of these will be unnecessary if I can just use `ghcr.io/dita-ot/dita-ot/dita-ot:3.6` as a base image.

- `openjdk-11-jre` - present in `adoptopenjdk:11-jre-hotspot`
-  `ant` - not needed in the base image (Unit test coverage requirement)
-  `curl` - present in adoptopenjdk:11-jre-hotspot
-  `unzip` -  present in `ghcr.io/dita-ot/dita-ot/dita-ot:3.6`
-  `git` - not needed in the base image (Unit test coverage requirement)
- `locales`  - **added**
- `tzdata`   - **added**

I've discovered that `locales`  is necessary in the following weird edge case:

```xml
<ph props="phoneme alphabet(ipa) ph(t&#x259;mei&#x27E;ou&#x325;)">tomato</ph>
	This example shows how a pronunciation for "tomato" is specified using the SPR phonology:
```

I want this to display as International Phonetic Alphabet:

```xml
 <phoneme alphabet="ipa" ph="təmeiɾou̥">tomato </phoneme>
```

However the default in the docker image is `LC_POSIX_UNIX` and that doesn't encode these UTF-8 chars properly. Changing the locale to `en_US.UTF-8` fixes the issue. Now I'm not saying the base image should set the locale directly, but it would be very useful to include the tools to do so for other images downstream without needing to use `apt-get` again e.g.:

```Dockerfile
RUN export DEBIAN_FRONTEND=noninteractive && \
	ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime && \
	dpkg-reconfigure --frontend noninteractive locales tzdata && \
	locale-gen en_US.UTF-8
```

Obviously I could equally want to change to French or German or Vietnamese or Finnish or whatever. `tzdata` is kind of a companion to `locales` as I may want to add date stamps according to my preferred local time.

There are other minor improvements included:

- The unzip is now less noisy.
- The base image currently needs a `build-arg` to work - defaults have been added.

The following build statement now works:

```console
docker build -t dita-ot .
```

- The base image currently has no metadata `LABEL` statements - these have added (see [opencontainers.org](https://github.com/opencontainers/opencontainers.org) )

To properly build a specific version and even add more metadata `LABELS` you can do something like this :

```console
docker build -t ghcr.io/dita-ot/dita-ot/dita-ot:3.6 --build-arg VERSION=3.6 \
        --label "org.opencontainers.image.created"=$(date +%Y-%m-%d) \
        --label "org.opencontainers.image.version"=3.6 .
docker push ghcr.io/dita-ot/dita-ot/dita-ot:3.6 
```

## How Has This Been Tested?

Build and run locally.

## Type of Changes

- Bug fix _(non-breaking change which fixes an issue)_
- New feature _(non-breaking change which adds functionality)_

## Documentation and Compatibility

Documentation is part of the discussion of #3664. 
This will reduce the proposed GitHub Container Action Dockerfile to:

```Dockerfile
FROM ghcr.io/dita-ot/dita-ot/dita-ot:3.6

SHELL ["/bin/bash", "-o", "pipefail", "-c"]

USER root
WORKDIR /

COPY entrypoint.sh entrypoint.sh
COPY script.sh install_script.sh
COPY script.sh build_script.sh

RUN export DEBIAN_FRONTEND=noninteractive && \
	ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime && \
	dpkg-reconfigure --frontend noninteractive locales tzdata && \
	locale-gen en_US.UTF-8 && \
    chmod +x /entrypoint.sh && \
	chmod +x /build_script.sh  && \
	chmod +x /install_script.sh

ENV LANG en_US.UTF-8  
ENV LANGUAGE en_US:en  
ENV LC_ALL en_US.UTF-8

ENTRYPOINT ["/entrypoint.sh"]
```

Which is basically just saying I'm US English (by default) and I need some bash scripts.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
-RUN apt-get update \
-    && apt-get install -y unzip \
-    && rm -rf /var/lib/apt/lists/*
+ARG VERSION=3.6
+ARG JAVA_VERSION=11-jre-hotspot
 
+FROM adoptopenjdk:${JAVA_VERSION}
 ARG VERSION
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
```
Comment (jelovirt) [inline Dockerfile:7]: Why is `-c` used here?

Comment (jelovirt) [inline Dockerfile:7]:
```suggestion
SHELL ["/bin/bash", "-euo", "pipefail", "-c"]
```

Adds additional error handling we could have

File: Dockerfile [hunk 2]
```diff
@@ -1,12 +1,27 @@
-FROM adoptopenjdk:11-jre-hotspot
-
-RUN apt-get update \
-    && apt-get install -y unzip \
-    && rm -rf /var/lib/apt/lists/*
+ARG VERSION=3.6
+ARG JAVA_VERSION=11-jre-hotspot
 
+FROM adoptopenjdk:${JAVA_VERSION}
```
Comment (jelovirt) [inline Dockerfile:4]: Not sure this is really needed.

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,27 @@
-FROM adoptopenjdk:11-jre-hotspot
-
-RUN apt-get update \
-    && apt-get install -y unzip \
-    && rm -rf /var/lib/apt/lists/*
+ARG VERSION=3.6
```
Comment (jelovirt) [inline Dockerfile:1]: Defaulting the version means we we would have to update this file every time we release a new version. My current preference is to not have a default for this.

File: Dockerfile [hunk 4]
```diff
@@ -1,12 +1,27 @@
...
+ARG JAVA_VERSION=11-jre-hotspot
 
+FROM adoptopenjdk:${JAVA_VERSION}
 ARG VERSION
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+LABEL "maintainer"="DITA Open Toolkit Development Team"
```
Comment (jelovirt) [inline Dockerfile:9]: There is no _DTA Open Toolkit Development Team_, so I'm not sure what this should be.

File: Dockerfile [hunk 5]
```diff
@@ -1,12 +1,27 @@
...
+LABEL "org.opencontainers.image.authors"="Robert D Anderson, Jarno Elovirta, Roger W. Fienhold Sheen"
```
Comment (jelovirt) [inline Dockerfile:10]: My preference would be to not put our names here, just because none of us wants people contacting us directly.

File: Dockerfile [hunk 6]
```diff
@@ -1,12 +1,27 @@
...
+LABEL "org.opencontainers.image.documentation"="https://www.dita-ot.org/"
+LABEL "org.opencontainers.image.vendor"="DITA Open Toolkit Development Team"
+LABEL "org.opencontainers.image.licenses"="Apache-2.0"
+LABEL "org.opencontainers.image.title"="DITA Open Toolkit"
+LABEL "org.opencontainers.image.description"="The open-source XML publishing engine for content authored in the Darwin Information Typing Architecture."
```
Comment (jelovirt) [inline Dockerfile:15]:
```suggestion
LABEL "org.opencontainers.image.description"="Publishing engine for content authored in the Darwin Information Typing Architecture."
```

File: Dockerfile [hunk 7]
```diff
@@ -26,4 +41,4 @@ USER dita-ot
 ENV DITA_HOME=/opt/app
 ENV PATH=${PATH}:${DITA_HOME}/bin
 WORKDIR $DITA_HOME
-ENTRYPOINT ["/opt/app/bin/dita"]
+ENTRYPOINT ["/opt/app/bin/dita"]
```
Comment (jelovirt) [inline Dockerfile:44]: Add missing line feed back

**Review by jelovirt — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (jelovirt) [reply Dockerfile:7]: Maybe it's safer to not add `-eu`, because those might be too strict.

File: Dockerfile (hunk 4 above)
Comment (jason-fox) [reply Dockerfile:9]: Typically this would be the legal entity that owns the copyright as found in the LICENCE file - The DITA-OT LICENSE hasn't listed the owner explicitly, so I would assume that the contributors to the repository - any suggestion as to how to rephrase it?

File: Dockerfile (hunk 5 above)
Comment (jason-fox) [reply Dockerfile:10]: Taken from `https://www.dita-ot.org/who_we_are` - I'll replace with the link.

File: Dockerfile (hunk 3 above)
Comment (jason-fox) [reply Dockerfile]: Reverted - 4b7c819

File: Dockerfile (hunk 2 above)
Comment (jason-fox) [reply Dockerfile]: Reverted - 4b7c819

File: Dockerfile (hunk 1 above)
Comment (jason-fox) [reply Dockerfile:4]: This `SHELL` is sanity checking, if we're going to fail let's fail hard. See https://github.com/docker/for-linux/issues/408

Comment (jason-fox) [reply Dockerfile:4]: I have always used `SHELL ["/bin/bash", "-o", "pipefail", "-c"]` - I think Hadolint prefers it as well.

File: Dockerfile (hunk 4 above)
Comment (infotexture) [reply Dockerfile:6]: In the docs, we use _“DITA Open Toolkit project”_ when we need a collective noun for the team. 

For [example](https://www.dita-ot.org/dev/topics/plugin-coding-conventions.html):

> … the DITA Open Toolkit project recommends …

File: Dockerfile (hunk 1 above)
Comment (jelovirt) [reply Dockerfile:4]: `-o pipefail` returns the first non-zero exit code in a pipe, but what about `-c`. The manual says

> -c
>    Read and execute commands from the first non-option argument command_string, then exit. If there are arguments after the command_string, the first argument is assigned to $0 and any remaining arguments are assigned to the positional parameters. The assignment to $0 sets the name of the shell, which is used in warning and error messages.

I have trouble parsing that. Could you elaborate what change it introduces?

File: Dockerfile (hunk 4 above)
Comment (jelovirt) [reply Dockerfile:6]: _DITA Open Toolkit project_ is good, since the maintainers really are just the people who file pull requests and people can come and go.

File: Dockerfile (hunk 1 above)
Comment (jason-fox) [reply Dockerfile:4]: Well you ain't going to need `-c` until you need it. It really only saves you from some weird edge cases. It helps with properly unquoting the command line parameters, say you want to invoke the container passing in the transtype which has been read in using some elaborate bash conversion function or something.

The best explanation I could find was here: https://ask.metafilter.com/316796/Using-piping-with-nix-find-without-getting-mired-in-string-escaping

File: Dockerfile (hunk 4 above)
Comment (jason-fox) [reply Dockerfile]: _DITA Open Toolkit project_ it is then. 50b6181

**Comment by jason-fox** [comment]
Hmmm. I've just had a thought, maybe switching from `LC_POSIX_UNIX` to `en_US.UTF-8` actually  *is* necessary. What about indexing?

Take the following list - `bailey', 'boffey', 'böhm', 'brown'` - in the `en` locale these would sort ignoring diacritics using the following:

```text
bailey: 2927373d2f57010a010a
boffey: 294331312f57010a010a
böhm:   2943353f01859d060109
brown:  294943534101090109
```

But in `LC_POSIX_UNIX` any special chars will sort after standard ASCII, so `böhm` will come after `brown`.

What do you think?

File: Dockerfile (hunk 1 above)
Comment (jelovirt) [reply Dockerfile:4]: A concrete example would be that now you can run

```bash
$ docker run docker.pkg.github.com/dita-ot/dita-ot/dita-ot:3.6 install foo
```

and with the `-c` shell you can both

```
```bash
$ docker run docker.pkg.github.com/dita-ot/dita-ot/dita-ot:3.6 install foo
$ docker run docker.pkg.github.com/dita-ot/dita-ot/dita-ot:3.6 "install foo"
```

Seems useful.

**Review by jelovirt — APPROVED** [review]
